### PR TITLE
[Tests-Only] API test for upload file with Mtime

### PR DESCRIPTION
**Pull request by Talank**
## Description
API test for upload a file with Mtime header.

## Related Issue
https://github.com/owncloud/ocis-reva/issues/273

## How Has This Been Tested?
- locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -2058,6 +2075,48 @@ public function userUploadsAFileWithContentTo(
 		return $this->uploadFileWithContent($user, $content, $destination);
 	}
 
+	/**
+	 * @When user :user uploads file to :destination with mtime :mtime using the WebDAV API
+	 *
+	 * @param string $user
+	 * @param string $destination
+	 * @param string $mtime API uses mtime as time in millisecond since epoch
+	 * 						Human readable format is taken as input and then converted into milliseconds
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebDav.php:2084]: wrong indent

File: tests/acceptance/features/apiWebdavUpload1/uploadFile.feature [hunk 2]
```diff
@@ -97,3 +97,14 @@ Feature: upload file
...
+  Scenario Outline: upload file with mtime
+    Given using <dav_version> DAV path
+    When user "Alice" uploads file to "file.txt" with mtime "Thu, 08 Aug 2019 04:18:13 GMT" using the WebDAV API
+    Then as "Alice" file "file.txt" should exist
+    And the HTTP status code should be "201"
+    And as "Alice" the mtime of the file "file.txt" should be "Thu, 08 Aug 2019 04:18:13 GMT"
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavUpload1/uploadFile.feature:110]: more tests:
- upload into a folder
- move file after upload should not change mtime

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiWebdavUpload1/uploadFile.feature [hunk 3]
```diff
@@ -97,3 +97,14 @@ Feature: upload file
       | new         | /upload.1     | abc.txt     |
       | new         | /upload...1.. | abc...txt.. |
       | new         | /...          | ...         |
+
+  Scenario Outline: upload file with mtime
```
Comment (individual-it) [inline tests/acceptance/features/apiWebdavUpload1/uploadFile.feature:101]: you will have to skip the tests on OCIS for now, as the functionality is in a PR only

Comment (PVince81) [reply tests/acceptance/features/apiWebdavUpload1/uploadFile.feature:101]: it's been merged and will come with the next ocis-reva update https://github.com/owncloud/ocis-reva/pull/279

Comment (PVince81) [reply tests/acceptance/features/apiWebdavUpload1/uploadFile.feature:101]: let's skip it still, tag it with https://github.com/owncloud/ocis-reva/issues/174

**Review by individual-it — APPROVED** [review]

File: tests/acceptance/features/apiWebdavUpload1/uploadFile.feature [hunk 4]
```diff
@@ -97,3 +97,42 @@ Feature: upload file
       | new         | /upload.1     | abc.txt     |
       | new         | /upload...1.. | abc...txt.. |
       | new         | /...          | ...         |
+
+  @skipOnOcis
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavUpload1/uploadFile.feature:101]:
```suggestion
  @skipOnOcis owncloud/ocis-reva#174
```

File: tests/acceptance/features/apiWebdavUpload1/uploadFile.feature [hunk 5]
```diff
@@ -97,3 +97,42 @@ Feature: upload file
...
+    When user "Alice" uploads file to "file.txt" with mtime "Thu, 08 Aug 2019 04:18:13 GMT" using the WebDAV API
+    Then as "Alice" file "file.txt" should exist
+    And the HTTP status code should be "201"
+    And as "Alice" the mtime of the file "file.txt" should be "Thu, 08 Aug 2019 04:18:13 GMT"
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+  @skipOnOcis
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavUpload1/uploadFile.feature:113]: issue tag
```suggestion
  @skipOnOcis owncloud/ocis-reva#174
```

File: tests/acceptance/features/apiWebdavUpload1/uploadFile.feature [hunk 6]
```diff
@@ -97,3 +97,42 @@ Feature: upload file
...
+    When user "Alice" uploads file to "/testFolder/file.txt" with mtime "Thu, 08 Aug 2019 04:18:13 GMT" using the WebDAV API
+    Then as "Alice" file "/testFolder/file.txt" should exist
+    And the HTTP status code should be "201"
+    And as "Alice" the mtime of the file "/testFolder/file.txt" should be "Thu, 08 Aug 2019 04:18:13 GMT"
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+  @skipOnOcis
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavUpload1/uploadFile.feature:126]: same here

File: tests/acceptance/features/apiWebdavUpload1/uploadFile.feature (hunk 4 above)
Comment (individual-it) [reply tests/acceptance/features/apiWebdavUpload1/uploadFile.feature]: The format it @issue-ocis-reva-174

**Pull request merged**